eventkit_cloud/tasks/task_factory.py

```diff
@@ -28,21 +28,51 @@ def __init__(self, job_uid):
 
     def parse_tasks(self):
         if self.run:
-            provider_tasks = [provider_task for provider_task in self.job.provider_tasks.all()]
+            # provider_tasks = [provider_task for provider_task in self.job.provider_tasks.all()]
+            osm_task = None
+            osm_types = {'osm': None, 'osm-thematic': None}
+            provider_tasks = []
+            osm_provider_tasks = {}
+            for provider_task in self.job.provider_tasks.all():
+                provider_type = provider_task.provider.export_provider_type.type_name
+                if provider_type in ['osm', 'osm-thematic']:
+                    osm_types[provider_type] = True
+                    osm_provider_tasks[provider_type] = provider_task
+                else:
+                    provider_tasks.append(provider_task)
+            if osm_types.get('osm-thematic'):
+                provider_tasks.append(osm_provider_tasks.get('osm-thematic'))
+                osm_task = osm_provider_tasks.get('osm-thematic')
+            elif osm_types.get('osm'):
+                provider_tasks.append(osm_provider_tasks.get('osm'))
+                osm_task = osm_provider_tasks.get('osm')
+
             if provider_tasks:
                 header_tasks = []
                 for provider_task in provider_tasks:
                     # Create an instance of a task runner based on the type name
                     if self.type_task_map.get(provider_task.provider.export_provider_type.type_name):
                         task_runner = self.type_task_map.get(provider_task.provider.export_provider_type.type_name)()
                         os.makedirs(os.path.join(self.stage_dir, provider_task.provider.slug), 6600)
-                        export_provider_task_uid, task_runner_tasks = task_runner.run_task(user=self.job.user,
-                                                                                           provider_task_uid=provider_task.uid,
-                                                                                           run=self.run,
-                                                                                           stage_dir=os.path.join(
-                                                                                               self.stage_dir,
-                                                                                               provider_task.provider.slug),
-                                                                                           service_type=provider_task.provider.export_provider_type.type_name)
+                        if provider_task == osm_task:
```

Comment:
Lets add a comment here explaining what is going on, they look really similar and because the task_runner is an abstraction it is hard to trace back the reason for this.